src/components/Tooltip/types.ts

```diff
@@ -25,16 +27,59 @@ type TooltipProps = ChildrenProps & {
     /** Unique key of renderTooltipContent to rerender the tooltip when one of the key changes */
     renderTooltipContentKey?: string[];
 
+    /** Whether to left align the tooltip relative to wrapped component */
+    shouldForceRenderingLeft?: boolean;
+
+    shouldForceRenderingBelow?: boolean;
+
+    /** Additional styles for tooltip wrapper view */
+    wrapperStyle?: StyleProp<ViewStyle>;
+};
+
+type TooltipRect = {
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+};
+
+type BaseTooltipState = {
+    /** Is tooltip visible */
+    isVisible: boolean;
+
+    /** Show tooltip */
+    showTooltip: () => void;
+
+    /** Hide tooltip */
+    hideTooltip: () => void;
+
+    /** Update the tooltip bounding rectangle */
```

Comment:
> /** Update the **tooltip's target** bounding rectangle */

This is a big deal! We could even rename `updateBounds` to `updateTargetBounds`.

This is really confusing, and I'm the one who introduced the "target" naming as an attempt to organize this somehow.

There are three crucial elements of a tooltip in Expensify:
- tooltip's target, the app's element that _has_ a tooltip (or at least _can_ have a tooltip)
  - on `main`, we wrap the target with a `Tooltip` component
- the actual tooltip, that has two parts:
   - the wrapper
   - the content
 
**We measure all of these elements**, for different reasons.

![image](https://github.com/Expensify/App/assets/2590174/339c8480-d5c9-4585-b6a8-f0c7369705c8)